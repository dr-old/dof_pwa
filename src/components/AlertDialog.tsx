import * as React from "react";
import Button from "@mui/material/Button";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogTitle from "@mui/material/DialogTitle";
import { useTheme } from "@mui/material";

interface AlertDialogProps {
  open: boolean;
  onClose: () => void;
  title: string;
  content: string;
  onConfirm?: () => void; // Optional confirm action
  confirmText?: string;
  cancelText?: string;
}

const AlertDialog: React.FC<AlertDialogProps> = ({
  open,
  onClose,
  title,
  content,
  onConfirm,
  confirmText,
  cancelText,
}) => {
  const theme = useTheme();

  return (
    <Dialog
      open={open}
      onClose={onClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      PaperProps={{
        sx: {
          borderRadius: theme.shape.borderRadius * 1, // Customize border radius
        },
      }}>
      <DialogTitle
        id="alert-dialog-title"
        sx={{
          fontSize: "1.25rem", // Customize title font size
          fontWeight: 600,
        }}>
        {title}
      </DialogTitle>
      <DialogContent>
        <DialogContentText
          id="alert-dialog-description"
          sx={{
            fontSize: "0.85rem", // Customize content font size
          }}>
          {content}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose}>{cancelText}</Button>
        {onConfirm && (
          <Button onClick={onConfirm} autoFocus>
            {confirmText}
          </Button>
        )}
      </DialogActions>
    </Dialog>
  );
};

export default AlertDialog;
