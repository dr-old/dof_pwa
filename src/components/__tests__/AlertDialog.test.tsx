import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import AlertDialog from "../AlertDialog";

describe("AlertDialog Component", () => {
  test("renders AlertDialog correctly", () => {
    const { asFragment } = render(
      <AlertDialog
        open={true}
        onClose={() => {}}
        title="Confirm Action"
        content="Are you sure you want to proceed?"
        confirmText="Yes"
        cancelText="No"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays title and content correctly", () => {
    render(
      <AlertDialog
        open={true}
        onClose={() => {}}
        title="Confirm Action"
        content="Are you sure you want to proceed?"
        confirmText="Yes"
        cancelText="No"
      />
    );

    expect(screen.getByText("Confirm Action")).toBeInTheDocument();
    expect(
      screen.getByText("Are you sure you want to proceed?")
    ).toBeInTheDocument();
  });

  test("calls onClose when cancel button is clicked", () => {
    const onCloseMock = jest.fn();

    render(
      <AlertDialog
        open={true}
        onClose={onCloseMock}
        title="Confirm Action"
        content="Are you sure you want to proceed?"
        confirmText="Yes"
        cancelText="No"
      />
    );

    fireEvent.click(screen.getByText("No"));
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  test("calls onConfirm when confirm button is clicked", () => {
    const onConfirmMock = jest.fn();

    render(
      <AlertDialog
        open={true}
        onClose={() => {}}
        title="Confirm Action"
        content="Are you sure you want to proceed?"
        onConfirm={onConfirmMock}
        confirmText="Yes"
        cancelText="No"
      />
    );

    fireEvent.click(screen.getByText("Yes"));
    expect(onConfirmMock).toHaveBeenCalledTimes(1);
  });
});
