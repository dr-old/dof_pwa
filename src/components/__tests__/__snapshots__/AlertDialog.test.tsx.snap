// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog Component renders AlertDialog correctly 1`] = `<DocumentFragment />`;
